kodi_addon_checker/check_addon.py

```diff
@@ -302,3 +308,46 @@ def _check_file_whitelist(report: Report, file_index, addon_path):
 def relative_path(file_path):
     path_to_print = file_path[len(REL_PATH):]
     return ".{}".format(path_to_print)
+
+
+def _get_addons(xml_url):
+    """
+        addon.xml for the target Kodi version
+    """
+    content = requests.get(xml_url).content
+    tree = ET.fromstring(content)
+
+    return {
+        a.get("id"): a.get("version")
+        for a in tree.findall("addon")
+    }
+
+
+def _get_users_dependencies(addon_path):
+    """
+        User's addon.xml from pull request
+    """
+    addon_xml_path = os.path.join(addon_path, "addon.xml")
+
+    tree = ET.parse(addon_xml_path).getroot()
+
+    return {
+        i.get("addon"): i.get("version")
+        for i in tree.findall("requires/import")
+    }
+
+
+def _check_dependencies(report: Report, addon_path, repo_addons):
+    deps = _get_users_dependencies(addon_path)
+    ignore = ['xbmc.json', 'xbmc.gui', 'xbmc.json', 'xbmc.metadata', 'xbmc.python']
+
+    for required_addon, required_version in deps.items():
+        if required_addon not in repo_addons:
+            if required_addon not in ignore:
+                report.add(Record(WARNING, "Required addon %s not available in current repository." % required_addon))
+        else:
+            available_version = repo_addons[required_addon]
+
+            if LooseVersion(available_version) != LooseVersion(required_version) and (required_addon not in ignore):
+                report.add(Record(WARNING, "Version mismatch for addon %s. Required: %s, Available: %s "
```

Comment:
This should be a `PROBLEM` not a `WARNING`

Reply:
should [Line 347](https://github.com/xbmc/addon-check/pull/63/files#diff-0908daa4913eca95dda1221fb9159098R347) be a `PROBLEM` or should I leave it as a `WARNING
`

Reply:
Yes, absolutely. Sorry for missing that.